File: app/src/main/java/eu/kanade/tachiyomi/data/io/IOHandler.java
package eu.kanade.tachiyomi.data.io;

import android.content.Context;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

public class IOHandler {
    private static String getTempFilename(Context context) throws IOException {
        File outputDir = context.getCacheDir();
        File outputFile = File.createTempFile("temp_cover", "0", outputDir);
        return outputFile.getAbsolutePath();
    }

    public static String downloadMediaAndReturnPath(FileInputStream input, Context context) {
        FileOutputStream output = null;
        try {

            String tempFilename = getTempFilename(context);
            output = new FileOutputStream(tempFilename);

            int read;
            byte[] bytes = new byte[4096];
            while ((read = input.read(bytes)) != -1) {
                output.write(bytes, 0, read);
            }
            return tempFilename;
        } catch (IOException ignored) {
        } finally {
            if (input != null) try {
                input.close();
            } catch (Exception ignored) {
            }
            if (output != null) try {
                output.close();
            } catch (Exception ignored) {
            }
        }
        return null;

    }

}

File: app/src/main/java/eu/kanade/tachiyomi/data/io/IOHandler.kt
package eu.kanade.tachiyomi.data.io

import android.content.Context
import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException

/**
 * Returns temp file location.
 *
 * @param context context of application.
 * @throws IOException IO exception.
 * @return location of temp file.
 */
@Throws(IOException::class)
private fun getTempFilename(context: Context): String {
    // Get output directory.
    val outputDir = context.cacheDir

    // Create temporary file
    val outputFile = File.createTempFile("temp_cover", "0", outputDir)

    // Return path of temporary file
    return outputFile.absolutePath
}

/**
 * Download media to temp location and returns file path.
 *
 * @param input input stream containing input file.
 * @param context context of application.
 * @throws IOException IO exception.
 * @return location of temp file.
 */
@Throws(IOException::class)
fun downloadMediaAndReturnPath(input: FileInputStream, context: Context): String {
    var output: FileOutputStream? = null
    try {
        // Get temp file name.
        val tempFilename = getTempFilename(context)

        output = FileOutputStream(tempFilename)

        // Copy input stream to temp location.
        input.copyTo(output)

        return tempFilename
    } finally {
        // Close streams.
        input.close()
        output?.close()
    }

}